### PR TITLE
Update api version and product version in API folder

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/11107
https://github.com/open-cluster-management/backlog/issues/11005#event-4532279481

### CODE REVIEW

File: troubleshooting/trouble_reinstall.adoc [hunk 1]
```diff
@@ -28,7 +28,7 @@ If you have this problem, complete the following steps:
 ACM_NAMESPACE=<namespace>
 oc delete mch --all -n $ACM_NAMESPACE
 helm ls --namespace $ACM_NAMESPACE | cut -f 1 | tail -n +2 | xargs -n 1 helm delete --namespace $ACM_NAMESPACE
-oc delete apiservice v1beta1.webhook.certmanager.k8s.io v1.admission.cluster.open-cluster-management.io v1.admission.work.open-cluster-management.io
```
Comment (nathanweatherly) [inline troubleshooting/trouble_reinstall.adoc]: Please revert this change. This particular script is for cleaning up old installs (which would reference v1beta1)

**Review by cadawson — APPROVED** [review]
Looks reasonable to me. Thanks!

File: apis/application.json.adoc [hunk 2]
```diff
@@ -49,7 +49,7 @@ __URL__ : http://swagger.io
 [[_rhacm-docs_apis_application_jsoncreateapplication]]
 === Create a application
 ....
-POST /app.k8s.io/v1beta1/namespaces/{namespace}/applications
+POST /app.k8s.io/v1beta2/namespaces/{namespace}/applications
```
Comment (dockerymick) [inline apis/application.json.adoc]: @KevinFCormier Can you confirm that my update here is accurate? Should it be `v1` or `v1beta1`

File: apis/application.json.adoc [hunk 3]
```diff
@@ -128,7 +128,7 @@ __required__|Parameters describing the application to be created.|<<_rhacm-docs_
 [[_rhacm-docs_apis_application_jsonqueryapplications]]
 === Query all applications
 ....
-GET /app.k8s.io/v1beta1/namespaces/{namespace}/applications
+GET /app.k8s.io/v1beta2/namespaces/{namespace}/applications
```
Comment (dockerymick) [inline apis/application.json.adoc]: @KevinFCormier same question here

File: apis/application.json.adoc [hunk 4]
```diff
@@ -174,7 +174,7 @@ __required__|Namespace that you want to use, for example, default.|string
 [[_rhacm-docs_apis_application_jsonqueryapplication]]
 === Query a single application
 ....
-GET /app.k8s.io/v1beta1/namespaces/{namespace}/applications/{application_name}
+GET /app.k8s.io/v1beta2/namespaces/{namespace}/applications/{application_name}
```
Comment (dockerymick) [inline apis/application.json.adoc]: @KevinFCormier same question here

File: apis/application.json.adoc [hunk 5]
```diff
@@ -217,7 +217,7 @@ __required__|Namespace that you want to use, for example, default.|string
 [[_rhacm-docs_apis_application_jsondeleteapplication]]
 === Delete a application
 ....
-DELETE /app.k8s.io/v1beta1/namespaces/{namespace}/applications/{application_name}
+DELETE /app.k8s.io/v1beta2/namespaces/{namespace}/applications/{application_name}
```
Comment (dockerymick) [inline apis/application.json.adoc]: @KevinFCormier same question here

File: apis/clusterset.json.adoc [hunk 6]
```diff
@@ -141,7 +141,7 @@ __required__|Parameters describing the clusterset to be created.|<<_rhacm-docs_a
 [source,json]
 ----
 {
-  "apiVersion" : "cluster.open-cluster-management.io/v1alpha1",
+  "apiVersion" : "cluster.open-cluster-management.io/v1",
```
Comment (dockerymick) [inline apis/clusterset.json.adoc]: @jamestalton Can you confirm if my update here is accurate

File: apis/observability.json.adoc [hunk 7]
```diff
@@ -49,7 +49,7 @@ __URL__ : http://swagger.io
 [[_rhacm-docs_apis_multiclusterobservability_jsoncreatemulticlusterobservability]]
 === Create a multiclusterobservability resource
 ....
-POST /apis/observability.open-cluster-management.io/v1beta1/multiclusterobservabilities
+POST /apis/observability.open-cluster-management.io/v1beta2/multiclusterobservabilities
```
Comment (dockerymick) [inline apis/observability.json.adoc:52]: @clyang82 can you confirm that my change here is accurate? The update was made according to the suggestion from issue https://github.com/open-cluster-management/backlog/issues/11005#event-4532279481

File: apis/observability.json.adoc [hunk 8]
```diff
@@ -98,7 +98,7 @@ __required__|Parameters describing the MultiClusterObservability resource to be
 [source,json]
 ----
 {
-  "apiVersion": "observability.open-cluster-management.io/v1beta1",
+  "apiVersion": "observability.open-cluster-management.io/v1",
```
Comment (dockerymick) [inline apis/observability.json.adoc]: @clyang82 can you help confirm that this change is correct. I kept this as v1 based on your comment here: https://github.com/open-cluster-management/backlog/issues/9378#issuecomment-816408083

File: apis/policy.json.adoc [hunk 9]
```diff
@@ -100,7 +100,7 @@ __required__|Parameters describing the policy to be created.|<<_rhacm-docs_apis_
 [source,json]
 ----
 {
-  "apiVersion": "policy.open-cluster-management.io/v1/v1alpha1",
+  "apiVersion": "policy.open-cluster-management.io/v1",
```
Comment (dockerymick) [inline apis/policy.json.adoc:103]: @ycao56 can you confirm that my changes in for this API is correct?

File: applications/app_sample.adoc [hunk 10]
```diff
@@ -36,7 +36,7 @@ The following YAML structure shows the required fields for an application and so
 
 [source,yaml]
 ----
-apiVersion: app.k8s.io/v1beta1
+apiVersion: app.k8s.io/v1beta2
```
Comment (dockerymick) [inline applications/app_sample.adoc]: @birsanv May you confirm that my changes in this file are correct?

File: clusters/upgrade_cluster_discon.adoc [hunk 11]
```diff
@@ -177,7 +177,7 @@ When you finish deploying the {cincinnati-short} instance on your hub cluster, t
 . Paste the contents from your {cincinnati-short} instance. Your YAML instance might resemble the following manifest:
 +
 ----
-apiVersion: cincinnati.openshift.io/v1beta1
+apiVersion: cincinnati.openshift.io/v1beta2
```
Comment (dockerymick) [inline clusters/upgrade_cluster_discon.adoc:180]: @jamestalton may you also confirm that this change is correct too?

File: observability/observability_enable.adoc [hunk 12]
```diff
@@ -116,7 +116,7 @@ For more details, see https://access.redhat.com/documentation/en-us/red_hat_open
 View the following default YAML file for observability:
 +
 ----
-apiVersion: observability.open-cluster-management.io/v1beta1
+apiVersion: observability.open-cluster-management.io/v1beta2
```
Comment (dockerymick) [inline observability/observability_enable.adoc:128]: @clyang82 may I also receive confirmation for this change?

File: apis/application.json.adoc (hunk 2 above)
Comment (KevinFCormier) [reply apis/application.json.adoc]: There is no change for this API. `v1beta1` is still correct.

File: apis/application.json.adoc (hunk 3 above)
Comment (KevinFCormier) [reply apis/application.json.adoc]: There is no change for this API. `v1beta1` is still correct.

File: apis/application.json.adoc (hunk 4 above)
Comment (KevinFCormier) [reply apis/application.json.adoc]: There is no change for this API. `v1beta1` is still correct.

File: apis/application.json.adoc (hunk 5 above)
Comment (KevinFCormier) [reply apis/application.json.adoc]: There is no change for this API. `v1beta1` is still correct.

File: applications/app_sample.adoc (hunk 10 above)
Comment (KevinFCormier) [reply applications/app_sample.adoc]: There is no change for this API. `v1beta1` is still correct.

File: risk_compliance/gatekeeper_policy.adoc [hunk 13]
```diff
@@ -20,7 +20,7 @@ spec:
   object-templates:
     - complianceType: musthave
       objectDefinition:
-        apiVersion: templates.gatekeeper.sh/v1beta1
+        apiVersion: templates.gatekeeper.sh/v1beta2
```
Comment (dockerymick) [inline risk_compliance/gatekeeper_policy.adoc]: @ycao56 I see that v1beta1 is still used for gatekeeper policy in the policy-collection repo. Can you confirm that my change is accurate? Should this change be made at all? Should it be changed to `v1`?

File: troubleshooting/trouble_app_deploy.adoc [hunk 14]
```diff
@@ -11,20 +11,20 @@ If one or more of your application resources in the Subscription YAML file uses
 
 ----
 failed to install release: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for
-kind "Deployment" in version "extensions/v1beta1"
+kind "Deployment" in version "networking.k8s.io/v1"
```
Comment (dockerymick) [inline troubleshooting/trouble_app_deploy.adoc]: @KevinFCormier May you confirm that my change is accurate here. I made the update based on your comment: https://github.com/open-cluster-management/backlog/issues/9378#issuecomment-807222141

Comment (KevinFCormier) [reply troubleshooting/trouble_app_deploy.adoc]: This doc is giving an example of an API version that is deprecated, so it should remain as-is.

Just a note that for other cases, not everything in `extensions/v1beta1` moved to `networking.k8s.io/v1`. `Deployment` actually moved to `apps/v1`. (See https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) But we don't want to make that update here because this is specifically a troubleshooting statement about deprecated APIs.

File: apis/policy.json.adoc (hunk 9 above)
Comment (ycao56) [reply apis/policy.json.adoc:103]: This is correct

File: risk_compliance/gatekeeper_policy.adoc (hunk 13 above)
Comment (ycao56) [reply risk_compliance/gatekeeper_policy.adoc]: No. Please keep it as is

File: services/serv_overview.adoc [hunk 15]
```diff
@@ -43,7 +43,7 @@ Edit the `metadata` section of your ingress service definition to add the servic
 See the following sample of an Ingress service definition:
 
 ----
-apiVersion: extensions/v1beta1
+apiVersion: extensions/v1beta2
```
Comment (jnpacker) [inline services/serv_overview.adoc]: This should be updated to networking.k8s.io/v1 as well I believe.

File: apis/clusterset.json.adoc (hunk 6 above)
Comment (jamestalton) [reply apis/clusterset.json.adoc:144]: i believe so

File: clusters/upgrade_cluster_discon.adoc (hunk 11 above)
Comment (jamestalton) [reply clusters/upgrade_cluster_discon.adoc:180]: i believe so

File: apis/clusterset.json.adoc (hunk 6 above)
Comment (jamestalton) [reply apis/clusterset.json.adoc]: that would be a question for foundation

File: services/serv_overview.adoc (hunk 15 above)
Comment (dockerymick) [reply services/serv_overview.adoc]: Thanks Josh!

File: clusters/upgrade_cluster_discon.adoc (hunk 11 above)
Comment (dockerymick) [reply clusters/upgrade_cluster_discon.adoc:180]: Thanks James!

File: apis/observability.json.adoc (hunk 7 above)
Comment (dockerymick) [reply apis/observability.json.adoc:52]: Confirmed via Slack by @marcolan018 thank you!

File: observability/observability_enable.adoc (hunk 12 above)
Comment (dockerymick) [reply observability/observability_enable.adoc:128]: @birsanv may you confirm my suggested change? I was informed that this is related to application

Comment (dockerymick) [reply observability/observability_enable.adoc:128]: Informed by @birsanv that @xiangjingli might be the better contact. Xiangjing, may you confirm my suggested update?

Comment (xiangjingli) [reply observability/observability_enable.adoc:128]: This change is related to the observability api, which is not owned by app lifecycle. @clyang82  would be the best to confirm it

File: apis/clusterset.json.adoc (hunk 6 above)
Comment (dockerymick) [reply apis/clusterset.json.adoc]: @qiujian16 can you help confirm my proposed change?

Comment (qiujian16) [reply apis/clusterset.json.adoc]: clusterset api is still v1alpha1

Comment (dockerymick) [reply apis/clusterset.json.adoc]: @qiujian16 thanks for the confirmation!

File: apis/observability.json.adoc (hunk 8 above)
Comment (clyang82) [reply apis/observability.json.adoc]: Sorry for confusion. the `v1` version is for `CustomResourceDefinitions`. not for `observability`. Here the version should be `observability.open-cluster-management.io/v1beta2`

File: observability/observability_enable.adoc (hunk 12 above)
Comment (clyang82) [reply observability/observability_enable.adoc:128]: sorry for late. That is correct.  the version is  `observability.open-cluster-management.io/v1beta2`.

Comment (dockerymick) [reply observability/observability_enable.adoc:128]: No worries @clyang82 I didn't realize you all were on PTO. Thank you so much for still responding, I appreciate you

File: apis/observability.json.adoc (hunk 8 above)
Comment (dockerymick) [reply apis/observability.json.adoc]: Thanks for clarifying!

**Comment by dockerymick** [comment]
All comments and conflicts have been resolved. This PR is ready to be merged
